### ISSUE TITLE
Subtle graphical problem with scanlines overlay effect

### ISSUE THREAD

**Issue by M-a-r-k** [Rakashazi/emu-ex-plus-alpha#36]
I'm using a Huawei Y300 phone (Android 4.1.1, Adreno 203, 800x480 screen). I noticed a problem with the image when the scanlines overlay effect is enabled. This is with NES.emu, but probably applies to the other emulators. The effect is very subtle, it was just easiest to see with the blue background of Super Mario Bros.

It happens even with integer scaling and no linear filtering. If you imagine the NES display split into two triangles (lower-left and upper-right), the upper-right triangle shows a subtle glitch. See the attached image.

![screenshot_2015-07-05-07-13-17](https://cloud.githubusercontent.com/assets/2737716/8513460/dcb2213c-2363-11e5-9d83-a5993b65b2fc.png)

The problem doesn't happen when the scanlines overlay effect is disabled.

**Comment by rhester72** [comment]
The issue, I believe, is that the scanlines (or any other) effect is applied before the stretching to screen size.  Ideally, they would be applied as the last stage of the render pipeline (a la MAME).

**Comment by M-a-r-k** [comment]
I don't think this is the same issue. In this case there is no scaling (or exact integer scaling); the output vertical resolution is 480 lines. [The screenshot above may be being scaled by your web browser. Save it and load into a paint package to view at 100% or 200% etc.]

I took a screenshot on the other Android device I have, an Archos GamePad:
![gamepad_screenshot](https://cloud.githubusercontent.com/assets/2737716/8580625/89631318-25b3-11e5-9867-333900bc7677.png)
Again, save that image and load into a paint package, since your web browser will probably scale it.

On the GamePad the graphical defect is different. The strange pattern in the background may be more noticeable if you change the angle of your eyes relative to the screen. Or load the image into a paint package and zoom in.

I did that on the GamePad image and noticed the black lines (since that's what 100% scanlines effect should produce) are not actually black. The colour varies from pixel to pixel, from (R:G:B) 0:1:2 to 3:4:6. And similarly, as you move along a line of blue sky the pixel RGB values vary from one pixel to the next.

Looking at the Y300 phone image, even in the "good" lower/left part of the image, while all the black scanline pixels are actually black (R:G:B 0:0:0), on every other blue line one in every four pixels is 90:158:255, as compared to 90:150:255 for the rest of the pixels.

I wonder if there is something wrong with the way the emulator uses OpenGL, and the screenshot differences are down to differing OpenGL drivers. Since there is (or should be) no scaling occurring surely it should be possible to get OpenGL to not mess with the image?
